tests/test_signature.py

```diff
@@ -0,0 +1,390 @@
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+        
+class FakeSignatureAPI_Cat(Signature):
+    name = "FakeSignatureAPI_Cat"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    filter_apinames = set(["gethostbyname"])
+    filter_categories = set(["network"]) 
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+        
+class FakeSignatureAPI_Process(Signature):
+    name = "FakeSignatureAPI_Process"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    filter_apinames = set(["gethostbyname"])
+    filter_processnames = set(["powershell.exe"])
+     
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+
+class FakeSignatureCat_Process(Signature):
+    name = "FakeSignatureCat_Process"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True  
+
+    filter_processnames = set(["powershell.exe"])
+    filter_categories = set(["network"])
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+
+class FakeSignatureNonFiltered(Signature):
+    name = "FakeSignatureNonFiltered"
+    description = "Fake signature created for testing signatures triggering"
+    severity = 1
+    categories = ["malware"]
+    authors = ["@CybercentreCanada", "@cccs-mog"]
+    minimum = "1.3"
+    evented = True
+
+    filter_apinames = set(["gethostbyname"])
+    filter_processnames = set(["powershell.exe"])
+    filter_categories = set(["network"])    
+
+    def __init__(self, *args, **kwargs):
+        Signature.__init__(self, *args, **kwargs)
+        self.query = False
+
+    def on_call(self, call, process):
+        self.query = True
+
+    def on_complete(self):
+        if self.query:
+            return True
+
+
+class FakeSignatureFiltered(Signature):
```

Comment:
Small naming change for clarity

```suggestion
class FakeSignatureAPI_Cat_Process_With_OnCall_Check(Signature):
```